### PR TITLE
Convert multiple images into a Masonry Gallery block

### LINKED ISSUE #1753: Converting individual images into a Masonry Gallery will return a broken block
### Describe the bug:
If you select several images and try to convert them into a Masonry block, you will get this error.
Screen Shot on 2020-11-12 at 17:05:03

### To reproduce:

1. Add a several images blocks to a post
2. Select multiple images
3. Click on the block icon to convert them to a Masonry Block

### Expected behavior:
I would expect the different image blocks to be converted into a single Masonry Gallery Block.

### Screenshots:
Here is a GIF of the error in action:

ezgif com-gif-maker

### Isolating the problem:

- [x] This bug happens with no other plugins activated
- [x] This bug happens with a default WordPress theme active
- [x] This bug happens **without** (and with) the Gutenberg plugin active
- [x] I can reproduce this bug consistently using the steps above

The error was spotted on a WordPress.com site but I was able to reproduce this on a self hosted site, with OR without the Gutenberg plugin. No other plugins were active and the theme was TwentyTwenty.

### WordPress version:
WordPress 5.5.3

### Gutenberg version:
9.3.0

### Error log

Here is the error coming from the console:

> react-dom.min.de439aae.js?m=1599596311h:9 TypeError: Cannot read property 'align' of undefined
>      at ??-eJyd0EkOgzAMAMAPNYRFQuJQ9S1ZLBpIY2Q70P6+gUvPzdHbyLY+NuUwCSTRW8xzSKznXEILNJcKgZ6avum1zSF6vcLHoiGv+IkkLgvrkDy8m4Vv+j8qoQQH9fOCKyQVA0s1sQc4tnJHNXAgeYc51Qs2olsV+CBI1Yg3Yq5+wlj/0DOhTqpaIMhsbAR1XVW/CQPtQIqDB0VQkN9rHq97N7bD1Hbj0C1fc5/+Yw==:30
>      at lodash.ec373016.js?m=1599596311h&ver=4.17.19:41
>      at lodash.ec373016.js?m=1599596311h&ver=4.17.19:371
>      at baseForOwn (lodash.ec373016.js?m=1599596311h&ver=4.17.19:183)
>      at lodash.ec373016.js?m=1599596311h&ver=4.17.19:368
>      at baseReduce (lodash.ec373016.js?m=1599596311h&ver=4.17.19:41)
>      at reduce (lodash.ec373016.js?m=1599596311h&ver=4.17.19:663)
>      at ??-eJyd0EkOgzAMAMAPNYRFQuJQ9S1ZLBpIY2Q70P6+gUvPzdHbyLY+NuUwCSTRW8xzSKznXEILNJcKgZ6avum1zSF6vcLHoiGv+IkkLgvrkDy8m4Vv+j8qoQQH9fOCKyQVA0s1sQc4tnJHNXAgeYc51Qs2olsV+CBI1Yg3Yq5+wlj/0DOhTqpaIMhsbAR1XVW/CQPtQIqDB0VQkN9rHq97N7bD1Hbj0C1fc5/+Yw==:30
>      at Array.reduce ()
>      at Object.attributes (??-eJyd0EkOgzAMAMAPNYRFQuJQ9S1ZLBpIY2Q70P6+gUvPzdHbyLY+NuUwCSTRW8xzSKznXEILNJcKgZ6avum1zSF6vcLHoiGv+IkkLgvrkDy8m4Vv+j8qoQQH9fOCKyQVA0s1sQc4tnJHNXAgeYc51Qs2olsV+CBI1Yg3Yq5+wlj/0DOhTqpaIMhsbAR1XVW/CQPtQIqDB0VQkN9rHq97N7bD1Hbj0C1fc5/+Yw==:30)
>  Me @ react-dom.min.de439aae.js?m=1599596311h:9
>  Jh.f.componentDidCatch.c.callback @ react-dom.min.de439aae.js?m=1599596311h:10
>  Wg @ react-dom.min.de439aae.js?m=1599596311h:7
>  oj @ react-dom.min.de439aae.js?m=1599596311h:9
>  Aj @ react-dom.min.de439aae.js?m=1599596311h:12
>  unstable_runWithPriority @ react.min.e713ea3b.js?m=1599596311h:3
>  Da @ react-dom.min.de439aae.js?m=1599596311h:7
>  ab @ react-dom.min.de439aae.js?m=1599596311h:11
>  Te @ react-dom.min.de439aae.js?m=1599596311h:10
>  (anonymous) @ react-dom.min.de439aae.js?m=1599596311h:7
>  unstable_runWithPriority @ react.min.e713ea3b.js?m=1599596311h:3
>  Da @ react-dom.min.de439aae.js?m=1599596311h:7
>  Pg @ react-dom.min.de439aae.js?m=1599596311h:7
>  ha @ react-dom.min.de439aae.js?m=1599596311h:7
>  mk.Events.current @ react-dom.min.de439aae.js?m=1599596311h:14
>  Ei @ react-dom.min.de439aae.js?m=1599596311h:3

Along with a BUNCH of other errors:
Screen Shot on 2020-11-12 at 17:02:03

### PR DESCRIPTION
**Pull request by jrtashjian**
### Description
Resolves #1753. The ability to select multiple image blocks and tranform them into the Masonry gallery has been added. This has also been added to the Carousel and Stacked blocks.

I was unable to get this working on the Collage and Offset gallery blocks due to a separate issue with how we manage internal state and attributes when the component mounts. I believe we need to refactor this a bit to prevent mutating the attributes unless the user has actually made a change.

This is PR considered a bug fix because the ability was built into the transforms feature of blocks but our code was not handling it properly.

### Types of changes
Bug fix (non-breaking change which fixes an issue)

### How has this been tested?
Jest tests

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Comment by jrtashjian** [comment]
Fixed all tests except the ones related to the settings modal. Fixes for that group of tests will be included in a new PR which refactors the modal store.

**Review by AnthonyLedesma — DISMISSED** [review]

File: .dev/tests/cypress/helpers.js [hunk 1]
```diff
@@ -140,7 +140,9 @@ export function addBlockToPost( blockName, clearEditor = false ) {
 	}
 
 	if ( clearEditor ) {
-		clearBlocks();
+		cy.wrap( null ).then( () =>
```
Comment (AnthonyLedesma) [inline .dev/tests/cypress/helpers.js:143]: 😨

**Comment by AnthonyLedesma** [comment]
Everything is working great locally. I see you just pushed up a change for tests. Let's see how it plays out.

File: src/blocks/food-and-drinks/test/food-and-drinks.cypress.js [hunk 2]
```diff
@@ -62,7 +62,9 @@ describe( 'Block: Food and Drinks', function() {
 			cy.get( '[data-type="coblocks/food-item"]' ).first().find( '.block-editor-media-placeholder' ).should( 'not.exist' );
 		} );
 
-		helpers.selectBlock( 'food & drink' );
```
Comment (AnthonyLedesma) [inline src/blocks/food-and-drinks/test/food-and-drinks.cypress.js:65]: Why are we moving away from using helpers here?

Comment (jrtashjian) [reply src/blocks/food-and-drinks/test/food-and-drinks.cypress.js:65]: Good catch. I actually meant to update the helper function with the new method after verifying this new code was working.

Comment (jrtashjian) [reply src/blocks/food-and-drinks/test/food-and-drinks.cypress.js:65]: And apparently I didn't realize it is the way the helper function works... I'm not really sure why I did this now. Either way, I'm updating it to use the helper.

File: .dev/tests/cypress/helpers.js (hunk 1 above)
Comment (jrtashjian) [reply .dev/tests/cypress/helpers.js]: I'm going to revert this. Looks like what you tried didn't work either. I'm going to try simply creating a new page.

Comment (AnthonyLedesma) [reply .dev/tests/cypress/helpers.js]: Reverting sounds good. This is strange. I am confused as to why this condition is able to occur. It almost seems like it happens just when clearing the blocks.

**Review by AnthonyLedesma — APPROVED** [review]

**Pull request merged**